omnipaxos_storage/src/persistent_storage.rs

```diff
@@ -0,0 +1,384 @@
+        let serialized = entries
+            .into_iter()
+            .map(|entry| bincode::serialize(&entry).expect("Failed to serialize"));
+        let offset = self
+            .commitlog
+            .append(&mut MessageBuf::from_iter(serialized))
+            .expect("Falied to append entries");
+        offset.first() + offset.len() as u64
+    }
+
+    fn append_on_prefix(&mut self, from_idx: u64, entries: Vec<T>) -> u64 {
+        self.commitlog
+            .truncate(from_idx)
+            .expect("Failed to truncate log");
+        self.append_entries(entries)
+    }
+
+    fn get_entries(&self, from: u64, to: u64) -> Vec<T> {
+        let buffer = self
+            .commitlog
+            .read(from, ReadLimit::default())
+            .expect("Failed to read from log");
+        let mut entries = Vec::<T>::with_capacity((to - from) as usize);
+        let mut iter = buffer.iter();
+        for _ in from..to {
+            let msg = iter.next().unwrap();
+            entries.push(bincode::deserialize(msg.payload()).expect("Failed to deserialize"));
+        }
+        entries
+    }
+
+    fn get_log_len(&self) -> u64 {
+        self.commitlog.next_offset()
+    }
+
+    fn get_suffix(&self, from: u64) -> Vec<T> {
+        self.get_entries(from, self.commitlog.next_offset())
+    }
+
+    fn get_promise(&self) -> Ballot {
+        let value = self.rocksdb.get(NPROM).expect("Failed to retrive 'NPROM'");
+        match value {
+            Some(prom_bytes) => {
+                let b_store: BallotStorage =
+                    FromBytes::read_from(prom_bytes.as_slice()).expect("Failed to deserialize");
+                Ballot::with(b_store.n, b_store.priority, b_store.pid)
+            }
+            None => Ballot::default(),
+        }
+    }
+
+    fn set_promise(&mut self, n_prom: Ballot) {
+        let ballot_store = BallotStorage::with(n_prom);
+        let prom_bytes = AsBytes::as_bytes(&ballot_store);
+        self.rocksdb
+            .put(NPROM, prom_bytes)
+            .expect("Failed to set 'NPROM'")
+    }
+
+    fn get_decided_idx(&self) -> u64 {
+        let value = self
+            .rocksdb
+            .get(DECIDE)
+            .expect("Failed to retrive 'DECIDE'");
+        match value {
+            Some(ld_bytes) => {
+                FromBytes::read_from(ld_bytes.as_slice()).expect("Failed to deserialize")
+            }
+            None => 0,
+        }
+    }
+
+    fn set_decided_idx(&mut self, ld: u64) {
+        let ld_bytes = AsBytes::as_bytes(&ld);
+        self.rocksdb
+            .put(DECIDE, ld_bytes)
+            .expect("Failed to set 'DECIDE'");
+    }
+
+    fn get_accepted_round(&self) -> Ballot {
+        let value = self.rocksdb.get(ACC).expect("Failed to retrive 'ACC'");
+        match value {
+            Some(acc_bytes) => {
+                let b_store: BallotStorage =
+                    FromBytes::read_from(acc_bytes.as_slice()).expect("Failed to deserialize");
```

Comment:
Since the expect messages will be read by the user, it should be more clear what failed to deserialize and not just use the variable name (since they don't read the source code). **Please fix this for all expect messages in this file.**

```suggestion
                    FromBytes::read_from(acc_bytes.as_slice()).expect("Failed to deserialize the accepted ballot.");
```